### PR TITLE
Feature/table gateway params

### PR DESCRIPTION
**Pull request by AdamiecRadek**
ping @weierophinney 
Added optional parameter for selectWith function where you can provid e additional parameters to execute function.
This will allow to user tablegateway features with flexibility of ordinary select

### CODE REVIEW

File: src/TableGateway/AbstractTableGateway.php [hunk 1]
```diff
@@ -189,32 +191,35 @@ public function select($where = null)
...
     /**
      * @param Select $select
-     * @return null|ResultSetInterface
-     * @throws \RuntimeException
+     * @param array  $params - statement params
+     *
+     * @return null|\Zend\Db\ResultSet\ResultSetInterface
      */
-    public function selectWith(Select $select)
+    public function selectWith(Select $select, array $params = [])
```
Comment (alextech) [inline src/TableGateway/AbstractTableGateway.php]: Cannot make changes to functions without making corresponding changes to TableGatewayInterface signatures, where this is defined, and all other classes implementing it, if any.

File: src/TableGateway/AbstractTableGateway.php [hunk 2]
```diff
@@ -25,25 +25,25 @@
...
+ * @property int              $lastInsertValue
+ * @property string           $table
  */
 abstract class AbstractTableGateway implements TableGatewayInterface
 {
     /**
      * @var bool
      */
-    protected $isInitialized = false;
+    protected $isInitialized = FALSE;
```
Comment (zf2timo) [inline src/TableGateway/AbstractTableGateway.php]: @AdamiecRadek i think this shouldn't be changed. Also the other occurs of `null` and `false`

Comment (froschdesign) [inline src/TableGateway/AbstractTableGateway.php]: @zf2timo 
Right!

@AdamiecRadek 
Please undo all code styling changes.

File: src/TableGateway/AbstractTableGateway.php (hunk 1 above)
Comment (weierophinney) [inline src/TableGateway/AbstractTableGateway.php]: Not true. You can add optional arguments to implementations; it then becomes a matter of the user understanding whether or not they are using the interface signature, or the implementation signature. Considering the majority of TableGatewayInterface implementations start from this abstract class, it's fine — but we should likely add a todo item to the interface to add these new arguments in a new major version.

File: src/TableGateway/AbstractTableGateway.php (hunk 2 above)
Comment (AdamiecRadek) [inline src/TableGateway/AbstractTableGateway.php]: @froschdesign undone

**Comment by AGmakonts** [comment]
@weierophinney - Not directly related to this PR but I would like to know your opinion about this idea.

This PR was created to allow for usage of Table Gateway Features like master/slave with regular Select objects.

Why do those features need to be restricted to TG? Wouldn't it be better to have the option to set up additional features globally for an adapter? 

I know that master/slave feature requires the second adapter but possibly attaching slave adapter to the master is not such a bad idea.

Anyway..... I don't have any solutions to this yet but I this could be done I can start working on PR with those changes.

**Comment by ezimuel** [comment]
@AdamiecRadek unit tests are missing, can you provide it? Thanks.

**Comment by AdamiecRadek** [comment]
@ezimuel  Tests provided

**Comment by michalbundyra** [comment]
This repository has been moved to laminas/laminas-db. If you feel that this patch is still relevant, please re-open against that repository, and reference this issue. To re-open, we suggest the following workflow:
 - Squash all commits in your branch (`git rebase -i origin/{branch}`)
 - Make a note of all changed files (`git diff --name-only origin/{branch}...HEAD
 - Run the laminas/laminas-migration tool on the code.
 - Clone laminas/laminas-db to another directory.
 - Copy the files from the second bullet point to the clone of laminas/laminas-db.
 - In your clone of laminas/laminas-db, commit the files, push to your fork, and open the new PR.
 We will be providing tooling via laminas/laminas-migration soon to help  automate the process.

**Comment by michalbundyra** [comment]
This repository has been closed and moved to laminas/laminas-db; a new issue has been opened at https://github.com/laminas/laminas-db/issues/100.

**Pull request closed without merging**